modules/fedimint-ln-server/src/lib.rs

```diff
@@ -725,8 +725,17 @@ impl ServerModule for Lightning {
                 .await;
 
                 // TODO: sanity-check encrypted preimage size
-                dbtx.insert_new_entry(&OfferKey(offer.hash), &(*offer).clone())
-                    .await;
+                if dbtx
+                    .insert_entry(&OfferKey(offer.hash), &(*offer).clone())
+                    .await
+                    .is_some()
+                {
+                    // Technically the error isn't due to a duplicate encrypted preimage but due to
+                    // a duplicate payment hash, practically it's the same problem though: re-using
+                    // the invoice key. Since we can't eaily extend the error enum we just re-use
+                    // this variant.
+                    return Err(LightningOutputError::DuplicateEncryptedPreimage);
```

Comment:
I'm not suggesting a change, but curious if this single check is sufficient for handling contract offer malleability (https://github.com/fedimint/fedimint/issues/1397), making the previous `EncryptedPreimageIndexKey` check unnecessary.

Reply:
As discussed on the call yesterday:

* The encrypted preimage check is against submitting the same encrypted preimage with a lower amount again, cheating the recipient out of sats
* The payment hash check is to prevent someone from overwriting an existing offer

They are related, but not redundant.